packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -51,3 +62,235 @@ export async function fetchSnapshot(
         async () => epochTracker.fetchAndParseAsJSON<IOdspSnapshot>(url, { headers }, "snapshotTree"),
     );
 }
+
+export async function fetchLatestSnapshotCore(
+    odspResolvedUrl: IOdspResolvedUrl,
+    storageTokenFetcher: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    tokenFetchOptions: TokenFetchOptions,
+    snapshotOptions: ISnapshotOptions | undefined,
+    logger: ITelemetryLogger,
+    summarizerClient: boolean = false,
+    epochTracker: EpochTracker | undefined,
+    snapshotCacheEntry: IEntry,
+    persistedCache: IPersistedCache,
+): Promise<{
+    odspSnapshot: ISnapshotCacheValue,
+    cacheP?: Promise<void>,
+}> {
+    const snapshotUrl = odspResolvedUrl.endpoints.snapshotStorageUrl;
+    const url = `${snapshotUrl}/trees/latest?ump=1`;
+    const storageToken = await storageTokenFetcher(tokenFetchOptions, "TreesLatest");
+    assert(storageToken !== null, "Storage token should not be null");
+    const formBoundary = uuid();
+    const formParams: string[] = [];
+    formParams.push(`--${formBoundary}`);
+    formParams.push(`Authorization: Bearer ${storageToken}`);
+    formParams.push(`X-HTTP-Method-Override: GET`);
+    const logOptions = {};
+    if (snapshotOptions !== undefined) {
+        Object.entries(snapshotOptions).forEach(([key, value]) => {
+            if (value !== undefined) {
+                formParams.push(`${key}: ${value}`);
+                logOptions[`snapshotOption_${key}`] = value;
+            }
+        });
+    }
+    if (odspResolvedUrl.sharingLinkToRedeem) {
+        formParams.push(`sl: ${odspResolvedUrl.sharingLinkToRedeem}`);
+    }
+    formParams.push(`_post: 1`);
+    formParams.push(`\r\n--${formBoundary}--`);
+    const postBody = formParams.join("\r\n");
+    const headers: {[index: string]: any} = {
+        "Content-Type": `multipart/form-data;boundary=${formBoundary}`,
+    };
+
+    let controller: AbortController | undefined;
+    if (summarizerClient !== true && snapshotOptions?.timeout !== undefined) {
```

Comment:
I believe we can get rid of summarizerClient argument - it's not really needed.
Caller should either set or reset timeout as appropriate - that's sufficient.

Reply:
Done